controllers/storagecluster/reconcile.go

```diff
@@ -282,10 +275,132 @@ func (r *StorageClusterReconciler) validateStorageClusterSpec(instance *ocsv1.St
 	return nil
 }
 
+func (r *StorageClusterReconciler) reconcileDynamicWatches() error {
+
+	if err := r.reconcileVolumeGroupSnapshot(); err != nil {
+		return err
+	}
+
+	if err := r.reconcileOdfVolumeGroupSnapshot(); err != nil {
+		return err
+	}
```

Comment:
isn't the body of these funcs exactly same w/ only a change in argument? in client-op iirc it was like this due to adding indexes 🤔.

`reconcileCrdWatches(client.Object, name)`?